spec/menubar/utils_spec.lua

```diff
@@ -65,6 +65,65 @@ describe("menubar.utils lookup_icon_uncached", function()
         assert_found_in_path('awesome',  '/.icons/awesome/64x64/apps/awesome.png')
         assert_found_in_path('awesome2', '/.icons/awesome/scalable/apps/awesome2.png')
     end)
+
+-- Check for no argument :: if not icon_file or icon_file == "" then
+    it('no icon specified', function()
+        assert.is_false(utils.lookup_icon_uncached())
+        assert.is_false(utils.lookup_icon_uncached(nil)) -- Same as above?
+        assert.is_false(utils.lookup_icon_uncached(false))
+        assert.is_false(utils.lookup_icon_uncached(''))
+    end)
+
+-- Check for icon not in search path :: if icon_file:sub(1, 1) == '/' and supported_icon_formats[icon_file_ext] then
+    it('finds icons even those not in the search paths when full path specified', function()
+
+        -- Shimmed icon base directories contain the following icons:
+        --
+        --     usr/share/icon5.png
+        --     usr/share/icon6.xpm
+        --     usr/share/icon7.svg
+
+--        assert_found_in_path('/usr/share/icon5.png', '/usr/share/icon5.png')
```

Comment:
why those are commented out?

Reply:
As per my previous comments regarding local test cases that work but don't under Travis...
`ToDo?: 3 test lines to restore & resolve why they fail`

Reply:
My local tests (files exist)...
```
local utils = require("menubar.utils")
print("1>",utils.lookup_icon_uncached('/usr/share/icon5.png'))
print("2>",utils.lookup_icon_uncached('/usr/share/icon6.xpm'))
print("3>",utils.lookup_icon_uncached('/usr/share/icon7.svg'))
1>	/usr/share/icon5.png
2>	/usr/share/icon6.xpm
3>	/usr/share/icon7.svg
```

Reply:
yeah, that makes sense, there are no such icons under those filepaths like `/usr/share/icon5.png`

see setUp function of this test module to understand how filepath thing is mocked here

Reply:
Part of the PR adds the files

These other files that were added as part of the PR work...
```
        assert_found_in_path('icon6.xpm', '/.icons/icon6.xpm')
        assert_found_in_path('icon7.svg', '/.icons/icon7.svg')

```

Reply:
please read setup function aloud line by line and let me know if you still didn't understood something

Reply:
PS: The path is already in the setup() function and the files are added as part of the PR         ```
shim('get_system_data_dirs', {
            root .. "/usr/local/share",
            root .. "/usr/share"
})
```

Reply:
This where my understanding of Travis is limited. Doesn't isn't `local root = (os.getenv("SOURCE_DIRECTORY") or '.') .. "/spec/menubar'` and `root .. "/usr/share"`allow Travis to simulate  the '/usr/share' directory?

Please note that these work...
```
        assert_found_in_path('icon1', '/usr/share/pixmaps/icon1.png')
        assert_found_in_path('icon2', '/usr/share/icons/icon2.png')
```
As does...
`        assert_found_in_path('icon5.png', '/.icons/icon5.png')
`
Could you possibly copy icon5.png into https://github.com/awesomeWM/awesome/tree/master/spec/menubar/usr/share just to see if Travis is having trouble?

Reply:
travis is nothing to do with it, it seems you just have those files in /usr/share locally

Reply:
I added them as part of the PR...

![clipboard01](https://user-images.githubusercontent.com/43520033/51076697-eb2b7780-16ef-11e9-873a-fc434f2d582e.png)

Reply:
https://github.com/awesomeWM/awesome/pull/2542#discussion_r247314282
https://github.com/awesomeWM/awesome/pull/2542#discussion_r247314340

Reply:
Solved